### ISSUE TITLE
CheckboxInput does not display any InvalidHint to the user

### ISSUE THREAD

**Issue by r-bartolomei** [auth0/lock#2400]
### Checklist

- [X] I have looked into the [Readme](https://github.com/auth0/lock#readme) and [Examples](https://github.com/auth0/lock/blob/master/EXAMPLES.md), and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/lock/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

If you use `additionalSignUpFields` option with a `type: 'checkbox'` that has a `validator` function (https://auth0.com/docs/libraries/lock/lock-configuration#checkbox-field), the `src/ui/input/checkbox_input.jsx` is not wrapped within an `InputWrap` component but within a simple div and as a result, the `invalidHint` will never be displayed when you press on the Sign Up button.

### Reproduction

with the below code:
```javascript
new Auth0Lock(AUTH0_CLIENT_ID, AUTH0_DOMAIN, {
    container: LOCK_CONTAINER_ID,
    allowLogin: false,
    allowSignUp: true,
    initialScreen: 'signUp',
    mustAcceptTerms: true,
    additionalSignUpFields: [
        {
            type: 'checkbox',
            name: 'newsletter',
            prefill: 'false',
            placeholder: 'I hereby agree that I want to receive marketing emails from your company',
            validator: (value) => ({
                valid: value === 'true',
                hint: 'This is a mandatory field',
            }),
        },
    ],
});
```
because of ` valid: value === 'true',` the field becomes mandatory (the user needs to check it) before being able to finish the Sign Up.
But because of the missing `InputWrap` within the `CheckboxInput` (`src/ui/input/checkbox_input.jsx`), no error message will be shown to the user if the checkbox is not checked and the user will not know why he can't go through with the sign up process.

As a comparison, `src/ui/input/text_input.jsx` and `src/ui/input/select_input.jsx` have the elements wrapped within the `InputWrap` and those components are displaying the `invalidHint` correctly.

### Lock version

"auth0-lock": "12.0.2"

### Which browsers have you tested in?

Chrome

**Comment by ewanharris** [comment]
Hey @r-bartolomei, I just put up a PR that would solve this issue over at #2423. Would love to get your feedback on whether the current styling (checkbox text turned to red) is expected for you if there's an alternative.

**Comment by r-bartolomei** [comment]
Hey @ewanharris. If the red border is not possible then the checkbox text on default colour I think might look better because it would align with the rest of mandatory fields (if any) as they would not have the text content on red.

![image](https://github.com/auth0/lock/assets/115465962/eabff3c4-6709-47a7-918d-02a6946adbba)

**Comment by ewanharris** [comment]
Sorry for the delay @r-bartolomei, we've chatted about this and after a little investigation I think we might do the following:

1. If `invalidHint` is provided then we'll only make that text red
2. If `invalidHint` is not provided, then we'll make the `placeholder` content red as without that there is no other way to flag the required field

**Comment by ewanharris** [comment]
Hey @r-bartolomei, this change shipped today in version `12.2.0`, thanks for the request!
